x-pack/plugins/apm/server/lib/service_map/get_service_anomalies.ts

```diff
@@ -136,16 +138,19 @@ function transformResponseToServiceAnomalies(
 }
 
 export async function getMLJobIds(
-  ml: Required<Setup>['ml'],
+  ml: {
+    anomalyDetectors: ReturnType<MlPluginSetup['anomalyDetectorsProvider']>;
+  },
   environment?: string
 ) {
   const response = await getMlJobsWithAPMGroup(ml);
+
   // to filter out legacy jobs we are filtering by the existence of `apm_ml_version` in `custom_settings`
   // and checking that it is compatable.
   const mlJobs = response.jobs.filter(
     (job) => (job.custom_settings?.job_tags?.apm_ml_version ?? 0) >= 2
   );
-  if (environment) {
+  if (environment && environment !== ENVIRONMENT_ALL) {
```

Comment:
Was it a bug that we didn't take this into account previously?
also: this would go away if we got rid of "all" envs.

Reply:
This wasn't a bug before since the environment value was omitted (in the `uiFilters` object) when "All" was the selected environment filter. If we get rid of the "All" as a environment filter option from the alert creation overlay, then we can get rid of this. We could also omit this check if we do this check in the alert executor function up the call stack.